trieste/models/gpflow/sampler.py

```diff
@@ -174,7 +178,13 @@ def sample(self, at: TensorType, *, jitter: float = DEFAULTS.JITTER) -> TensorTy
         return mean[..., None, :, :] + tf.transpose(variance_contribution, new_order)
 
 
-class RandomFourierFeatureTrajectorySampler(TrajectorySampler):
+class SupportsGetKernelObservationNoise(SupportsGetKernel, SupportsGetObservationNoise, Protocol):
```

Comment:
I would count this as an interface and would move it to the interface file

Reply:
The reason I put it here is to avoid polluting the interface with arbitrary combinations of features. The features themselves all belong there obviously, but the combinations are more to do with the classes that require them I think.